src/Storageless/Http/Configuration.php

```diff
@@ -98,15 +98,15 @@ public function getClientFingerprintConfiguration(): FingerprintConfig
     public function withJwtConfiguration(JwtConfig $jwtConfiguration): self
     {
         $new                   = clone $this;
-        $new->jwtConfiguration = clone $jwtConfiguration;
+        $new->jwtConfiguration = $jwtConfiguration;
```

Comment:
This was on purpose: the cloning is specifically done to **NOT** have the `JwtConfig` change.

Reply:
Yeah, and that's wrong: mutability is part of an API, if an object is designed to be mutable we cannot violate its design.

We can only enforce immutability on our own library, not the others.

Reply:
Strong disagree: this is exactly why cloning is done here, to "detach" from the other system.

It was explicitly added here, on purpose.

Reply:
I'm not arguing the *why*: it has always been clear to me, and I agree with that.

I'm saying that even though we are capable of cloning external services, we shouldn't consider ourself allowed to.

A mutable dependency can contain all soft of interconnections with other services, and bad OOP practices.
We are type-hinting some interfaces, and it would be legit for me to feed a dependency that impements that interface and extends the `\Doctrine\ORM\UnitOfWork`: image the distaster if we clone such Kraken.

Reply:
Right, but we are cloning a config object here, not a UnitOfWork: dereferencing and common sense, please 😛

Let's revert this.

Reply:
Done :+1:

Reply:
Thanks!